module/zfs/zcp_get.c

```diff
@@ -404,6 +407,25 @@ get_special_prop(lua_State *state, dsl_dataset_t *ds, const char *dsname,
 		break;
 	}
 
+	case ZFS_PROP_ENCRYPTION_ROOT: {
+		setpoint[0] = '\0';
+		numval = 0;
```

Comment:
I wonder why do we need to initialize `numval` here?

And also wonder why you are not setting `setpoint` below, similar to above?

Reply:
`numval` is a leftover, can delete that;

filling `setpoint` with the path didn't work (the result I was getting was empty AFAIK) - I wasn't clear on what `setpoint` is for, intuitively I would say that it's relevant when the value is numerical or if there's the need to pass another string along; looking at `dsl_get_mountpoint`, its impl refers to the third parameter as `source`

originally I tried to lump `ZFS_PROP_ENCRYPTION_ROOT` with the rest of the pack above but those are all numeric values

if you check the other properties in this function above, some set only `strval` and only a few set both `setpoint` and `strval`

Reply:
@amotin while we're at this, what do you think of this:

- if you do `zfs get all`, then `encryptionroot` isn't listed there
- but `zfs get encryptionroot` works
- I know some properties might be intentionally hidden, those are AFAIK registered with the `zprop_register_hidden`  variant - but `ZFS_PROP_ENCRYPTION_ROOT` is registered with `zprop_register_string`, so that should be visible in `get all`, shouldn't it?

should I file an issue for that or am I missing something and it should in fact be hidden?

Reply:
I don't know why it is named so, but in the code above `setpoint` seems to be set to the property source, which is "default", "local", "received", etc.  And unless it somehow makes no sense for `ZFS_PROP_ENCRYPTION_ROOT`, it would be nice to report it too.

I am not aware why `ZFS_PROP_ENCRYPTION_ROOT` might be hidden.  I can think of two reasons to hide properties: either they are too technical to expose to user, but still somehow useful for some tools, or it is too expensive to fetch.  I don't think either applies to `ZFS_PROP_ENCRYPTION_ROOT`, but I might be wrong on the first side, since I am not too deep in encryption administration.

Reply:
ok will take another look at `setpoint`;

btw @ `zfs get all`, apparently a property that doesn't have a value at all isn't listed in `get all`, although it is accessible via direct get; if I do `get all` with a dataset that does have `encryptionroot`, it works

Reply:
As for why `ZFS_PROP_ENCRYPTION_ROOT` my recollection is at the time we felt it was too technical and it didn't need to be visible.  We might want to revisit that.  There isn't any issue in allowing it to be fetched.